Frends.SFTP.DownloadFiles/Frends.SFTP.DownloadFiles/Definitions/SingleFileTransfer.cs

```diff
@@ -1,500 +1,487 @@
+                    TransferState.DeleteDestinationFile,
+                    "Deleting destination file {0} that is to be overwritten",
                     Path.GetFileName(DestinationFileWithMacrosExpanded));
 
-                File.Move(DestinationFileDuringTransfer, DestinationFileWithMacrosExpanded);
+                File.Delete(DestinationFileWithMacrosExpanded);
             }
-        }
 
-        /// <summary>
-        /// Restores the LastWriteTime stamp to the destination file.
-        /// </summary>
-        private void RestoreModified()
-        {
-            DestinationFile.Modified = Client.GetAttributes(SourceFile.Name).LastWriteTime;
-            return;
+            Trace(
+                TransferState.RenameDestinationFile,
+                "Renaming temporary destination file {0} to target file {1}",
+                Path.GetFileName(DestinationFileDuringTransfer),
+                Path.GetFileName(DestinationFileWithMacrosExpanded));
+
+            File.Move(DestinationFileDuringTransfer, DestinationFileWithMacrosExpanded);
         }
+    }
+
+    /// <summary>
+    /// Restores the LastWriteTime stamp to the destination file.
+    /// </summary>
+    private void RestoreModified()
+    {
+        DestinationFile.Modified = Client.GetAttributes(SourceFile.FullPath).LastWriteTime;
+    }
 
-        private static Encoding GetEncoding(Destination dest)
+    private static Encoding GetEncoding(Destination dest)
+    {
+        switch (dest.FileNameEncoding)
         {
-            switch (dest.FileNameEncoding)
-            {
-                case FileEncoding.UTF8:
-                    return dest.EnableBomForFileName ? new UTF8Encoding(true) : new UTF8Encoding(false);
-                case FileEncoding.ASCII:
-                    return Encoding.ASCII;
-                case FileEncoding.ANSI:
-                    return Encoding.Default;
-                case FileEncoding.Unicode:
-                    return Encoding.Unicode;
-                case FileEncoding.WINDOWS1252:
-                    return Encoding.Default;
-                case FileEncoding.Other:
-                    return Encoding.GetEncoding(dest.FileNameEncodingInString);
-                default:
-                    throw new ArgumentOutOfRangeException();
-            }
+            case FileEncoding.UTF8:
+                return dest.EnableBomForFileName ? new UTF8Encoding(true) : new UTF8Encoding(false);
+            case FileEncoding.ASCII:
+                return Encoding.ASCII;
+            case FileEncoding.ANSI:
+                return Encoding.Default;
+            case FileEncoding.Unicode:
+                return Encoding.Unicode;
+            case FileEncoding.WINDOWS1252:
+                return Encoding.Default;
+            case FileEncoding.Other:
+                return Encoding.GetEncoding(dest.FileNameEncodingInString);
+            default:
+                throw new ArgumentOutOfRangeException();
```

Comment:
Maybe add some kind of error message to indicate what went wrong.

Reply:
Added error message